MetaMask/metamask-extension
# [e2e] Security provider improvements

## Explanation

**Context**: there has been an issue, where we sent additional requests to the Security Provider [#18560](https://github.com/MetaMask/metamask-extension/issues/18560) . This issue has been partially fixed by this PR [#18598](https://github.com/MetaMask/metamask-extension/pull/18598)

**What**: this PR adds new assertions to the e2e security provider tests, for making sure that only 1 request is sent to the security provider.
Since the additional requests were sent both whenever a signature was triggered as well as when the tx was Accepted/Rejected, in this PR we extend the actions to finish the flow, by accepting the signatures and at the end of the test we make sure only 1 request has been sent.

- Fixes https://github.com/MetaMask/metamask-extension/issues/18615


## Screenshots/Screencaps
Whenever the remaining issue is fixed, this PR should pull last develop and tests should pass.
At the moment, tests fail, meaning tests are catching the extra requests.

![Screenshot from 2023-04-17 11-19-16](https://user-images.githubusercontent.com/54408225/232441640-48329fae-7335-4d11-a189-39ccbf12e32d.png)


### Before

/
### After

/
## Manual Testing Steps
- Check that all tests pass in circle ci
- Manually run the security provider test in your local test environment `yarn test:e2e:single test/e2e/tests/security-provider.spec.js --browser=chrome`

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
